packages/integration-tests/test/challengeExitTest.js

```diff
@@ -0,0 +1,155 @@
+limitations under the License. */
+
+const config = require('../test-config')
+const helper = require('./helper')
+const Web3 = require('web3')
+const ChildChain = require('@omisego/omg-js-childchain')
+const RootChain = require('@omisego/omg-js-rootchain')
+const chai = require('chai')
+const assert = chai.assert
+
+const GETH_URL = `http://${config.geth.host}:${config.geth.port}`
+const web3 = new Web3(GETH_URL)
+const childChain = new ChildChain(`http://${config.watcher.host}:${config.watcher.port}`, `http://${config.childchain.host}:${config.childchain.port}`)
+const rootChain = new RootChain(GETH_URL, config.plasmaContract)
+const ETH_CURRENCY = '0000000000000000000000000000000000000000'
+
+// NB This test is designed to run against a modified RootChain contract that allows exits after 20 seconds.
+const CHALLENGE_PERIOD = 20 * 1000
+
+describe('Challenge exit tests', async () => {
+  const INTIIAL_ALICE_AMOUNT = web3.utils.toWei('2', 'ether')
+  const INTIIAL_BOB_AMOUNT = web3.utils.toWei('1', 'ether')
+  const DEPOSIT_AMOUNT = web3.utils.toWei('1', 'ether')
+  const TRANSFER_AMOUNT = web3.utils.toWei('0.2', 'ether')
+  let aliceAccount
+  let bobAccount
+
+  beforeEach(async () => {
+    const accounts = await web3.eth.getAccounts()
+    // Assume the funding account is accounts[0] and has a blank password
+    // Create and fund Alice's account
+    aliceAccount = await helper.createAndFundAccount(web3, accounts[0], '', INTIIAL_ALICE_AMOUNT)
+    console.log(`Created Alice account ${JSON.stringify(aliceAccount)}`)
+    // Create Bob's account
+    bobAccount = await helper.createAndFundAccount(web3, accounts[0], '', INTIIAL_BOB_AMOUNT)
+    console.log(`Created Bob account ${JSON.stringify(bobAccount)}`)
+    // Alice deposits ETH into the Plasma contract
+    await rootChain.depositEth(DEPOSIT_AMOUNT, aliceAccount.address, aliceAccount.privateKey)
+    // Wait for transaction to be mined
+    const balance = await helper.waitForBalance(childChain, aliceAccount.address, DEPOSIT_AMOUNT)
+    assert.equal(balance[0].amount.toString(), DEPOSIT_AMOUNT)
+    console.log(`Alice deposited ${DEPOSIT_AMOUNT} into RootChain contract`)
+  })
+
+  it('should succesfully challenge a dishonest exit', async () => {
+    // Send TRANSFER_AMOUNT from Alice to Bob
+    await helper.sendAndWait(childChain, aliceAccount.address, bobAccount.address, Number(TRANSFER_AMOUNT), [aliceAccount.privateKey], TRANSFER_AMOUNT)
+    console.log(`Transferred ${TRANSFER_AMOUNT} from Alice to Bob`)
+
+    // Save Alice's latest utxo
+    const aliceUtxos = await childChain.getUtxos(aliceAccount.address)
+    const aliceDishonestUtxo = aliceUtxos[0]
+
+    // Send another TRANSFER_AMOUNT from Alice to Bob
+    await helper.sendAndWait(childChain, aliceAccount.address, bobAccount.address, Number(TRANSFER_AMOUNT), [aliceAccount.privateKey], TRANSFER_AMOUNT * 2)
+    console.log(`Transferred ${TRANSFER_AMOUNT} from Alice to Bob again`)
```

Comment:
Why the second transfer?

Reply:
That's what makes Alice's exit dishonest - if she can successfully exit after the _first_ transfer then it's as if the _second_ transfer never happened, thus cheating Bob out of that second payment.

Reply:
Right, right. Got my utxos mixed up. Was thinking about exiting the deposit, which doesn't make sense in this order anyway.